### PR TITLE
Add keyboard navigation when the grid is focused

### LINKED ISSUE #606: Page Up/Page Down buttons don't scroll the grid
### Description  
Page Up/Page Down buttons don't scroll the grid

### Steps to reproduce  

1. Step 1
2. Step 2
3. ...

### Result  
Page Up/Page Down buttons don't scroll the grid

### Expected result  
Page Up/Page Down buttons should scroll the grid

### PR DESCRIPTION
**Pull request by PavlovVasil**
Closes #606  .  

The left/right arrow keys should scroll with the default minimum column width - 136px
The up/down arrow keys should scroll 1 row downwards/upwards
Pageup/Pagedown should scroll with one container size upwards/downwards

### CODE REVIEW

**Comment by MartinKamenov** [comment]
[ ] When pressed on grid's scroll it doesn't focus on the grid in IE.

**Comment by mpavlinov** [comment]
@MartinKamenov We will leave it as is.

File: projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts [hunk 1]
```diff
@@ -314,6 +314,14 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
         this.scrollTo(this.state.startIndex - 1);
     }
 
+    public scrollNextPage() {
+        this.vh.instance.elementRef.nativeElement.scrollTop += parseInt(this.igxForContainerSize, 10);
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts]: if this is public, it should work for horizontal virtualization as well, changing scrollLeft

**Comment by kdinev** [comment]
@MartinKamenov Moving this back to `status: awaiting-test` since it hasn't been tested with the latest code in `6.1.x`

**Comment by MartinKamenov** [comment]
@PavlovVasil When scrolling in grid with bigger data size using arrows at some point the scrolling stops.
#### Steps to reproduce
1. Open grid performance dev sample.
2. Focus on grid
3. Scroll down using down arrow button until you get to row with ID - 21.

#### Expected behaviour
The scroll should stop when scrolling too much.

File: projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts [hunk 2]
```diff
@@ -462,7 +462,6 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
         this.scrollTo(this.state.startIndex - 1);
     }
 
-<<<<<<< HEAD
     public scrollNextPage() {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts]: Add documentation.

File: projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts [hunk 3]
```diff
@@ -462,6 +462,23 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
         this.scrollTo(this.state.startIndex - 1);
     }
 
+    public scrollNextPage() {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts]: Add API docs to both new methods.

**Pull request merged**